### PR TITLE
Add functional tests

### PR DESCRIPTION
**Pull request by mkdesignn**
### Don't merge the pull-request

### CODE REVIEW

**Comment by mkdesignn** [comment]
### Don't merge the pull-request

File: pom.xml [hunk 1]
```diff
@@ -59,6 +52,25 @@
...
+<!--            <version>3.3.0-01</version>-->
+<!--        </dependency>-->
+
+<!--        <dependency>-->
+<!--            <groupId>org.codehaus.groovy</groupId>-->
+<!--            <artifactId>groovy-eclipse-batch</artifactId>-->
+<!--            <version>2.4.17-01</version>-->
+<!--        </dependency>-->
+
+
```
Comment (Imany-apk) [inline pom.xml]: what are these comments for?

File: pom.xml [hunk 2]
```diff
@@ -42,13 +42,6 @@
             <scope>test</scope>
         </dependency>
 
-        <dependency>
-            <groupId>org.spockframework</groupId>
-            <artifactId>spock-spring</artifactId>
-            <version>1.2-groovy-2.4</version>
-            <scope>test</scope>
-        </dependency>
-
```
Comment (Imany-apk) [inline pom.xml]: did you remove Spock?

File: src/test/java/com/example/jwt/JwtApplicationTests.groovy [hunk 3]
```diff
@@ -1,17 +1,21 @@
...
 class JwtApplicationTests extends Specification {
 
+    @Shared
+    def client = new RESTClient( "$SERVER_URL:$SERVER_PORT")
 
-    def "writing first tests with groovy using spock"() {
-        expect: 'mohammad' == 'mohammad'
+    def 'should return 200 code when used valid credentials' () {
+        when: 'login with invalid credentials'
+        def response = client.get( path : '/login' )
```
Comment (Imany-apk) [inline src/test/java/com/example/jwt/JwtApplicationTests.groovy]: where is posted credentials?
and why are you using GET for login?

File: src/test/java/com/example/jwt/JwtApplicationTests.groovy [hunk 4]
```diff
@@ -1,17 +1,21 @@
...
 import org.springframework.boot.test.context.SpringBootTest
-import org.springframework.test.web.servlet.MockMvc;
+import spock.lang.Shared;
 import spock.lang.Specification;
 
 @SpringBootTest
 class JwtApplicationTests extends Specification {
 
+    @Shared
+    def client = new RESTClient( "$SERVER_URL:$SERVER_PORT")
```
Comment (Imany-apk) [inline src/test/java/com/example/jwt/JwtApplicationTests.groovy]: Where is the RESTClient imported from?
and where the $SERVER_URL and $SERVER_PORT come from?

File: pom.xml (hunk 2 above)
Comment (mkdesignn) [reply pom.xml]: NO, i have removed the latest one, but above the removed code you can see its still has the spock plugin.

File: pom.xml (hunk 1 above)
Comment (mkdesignn) [reply pom.xml]: I have already told you the reason in telegram, and i have also commented on the pull request to not merge it. just pull the branch and starts working on it. i could not write a functional test. maybe you can try write a functional test for login

Comment (mkdesignn) [reply pom.xml]: I have already told you the reason in telegram, and i have also commented on the pull request to not merge it. just pull the branch and starts working on it. i could not write a functional test. maybe you can try write a functional test for login

Comment (Imany-apk) [reply pom.xml]: I didn't understand that, paste it one more time :)))

**Pull request closed without merging**